airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -154,18 +156,20 @@ protected void acceptTracked(AirbyteMessage message) throws Exception {
     }
 
     // ignore other message types.
-    final AirbyteStreamNameNamespacePair pair = AirbyteStreamNameNamespacePair.fromRecordMessage(message.getRecord());
+    AirbyteRecordMessage record = message.getRecord();
+    final AirbyteStreamNameNamespacePair pair = AirbyteStreamNameNamespacePair.fromRecordMessage(record);
     if (!pairs.contains(pair)) {
       throw new IllegalArgumentException(
           String.format("Message contained record from a stream that was not in the catalog. \ncatalog: %s , \nmessage: %s",
               Jsons.serialize(catalog), Jsons.serialize(message)));
     }
-    var data = Jsons.serialize(message.getRecord().getData());
+    var data = Jsons.serialize(record.getData());
     if (isValidRecord.apply(data)) {
       // TODO Truncate json data instead of throwing whole record away?
       // or should we upload it into a special rejected record table instead?
-      var serialisedMsg = Jsons.serialize(message);
-      pairToWriteBuffer.get(pair).offer(serialisedMsg.getBytes(Charsets.UTF_8));
+      Timestamp emittedAt = Timestamp.from(Instant.ofEpochMilli(record.getEmittedAt()));
+      RecordData recordData = new RecordData(data, emittedAt);
+      pairToWriteBuffer.get(pair).offer(Jsons.serialize(recordData).getBytes(StandardCharsets.UTF_8));
```

Comment:
the double serialization looks inefficient

Reply:
You are right. This is a quick and dirty version working off the previous implementation that was also double serialising. I'll clean this up to serialise to bytes directly later on.